packages/@ember/-internals/glimmer/lib/modifiers/on.ts

```diff
@@ -0,0 +1,241 @@
+  passing `eventOptions`. In some situations it then throws a weird script
+  error, like:
+
+  ```
+  Could not complete the operation due to error 80020101
+  ```
+
+  This flag determines, whether `{ once: true }` and thus also event options in
+  general are supported.
+*/
+const SUPPORTS_EVENT_OPTIONS = (() => {
+  try {
+    const div = document.createElement('div');
+    let counter = 0;
+    div.addEventListener('click', () => counter++, { once: true });
+
+    let event;
+    if (typeof Event === 'function') {
+      event = new Event('click');
+    } else {
+      event = document.createEvent('Event');
+      event.initEvent('click', true, true);
+    }
+
+    div.dispatchEvent(event);
+    div.dispatchEvent(event);
+
+    return counter === 1;
+  } catch (error) {
+    return false;
+  }
+})();
+
+export class OnModifierState {
+  public tag: Tag;
+  public element: Element;
+  public args: CapturedArguments;
+  public eventName!: string;
+  public callback!: EventListener;
+  private userProvidedCallback!: EventListener;
+  public once?: boolean;
+  public passive?: boolean;
+  public capture?: boolean;
+  public options?: AddEventListenerOptions;
+  public shouldUpdate = true;
+
+  constructor(element: Element, args: CapturedArguments) {
+    this.element = element;
+    this.args = args;
+    this.tag = args.tag;
+  }
+
+  updateFromArgs() {
+    let { args } = this;
+
+    let { once, passive, capture }: AddEventListenerOptions = args.named.value();
+    if (once !== this.once) {
+      this.once = once;
+      this.shouldUpdate = true;
+    }
+
+    if (passive !== this.passive) {
+      this.passive = passive;
+      this.shouldUpdate = true;
+    }
+
+    if (capture !== this.capture) {
+      this.capture = capture;
+      this.shouldUpdate = true;
+    }
+
+    let options: AddEventListenerOptions;
+    if (once || passive || capture) {
+      options = this.options = { once, passive, capture };
+    } else {
+      this.options = undefined;
+    }
+
+    assert(
+      'You must pass a valid DOM event name as the first argument to the `on` modifier',
+      args.positional.at(0) !== undefined && typeof args.positional.at(0).value() === 'string'
+    );
+    let eventName = args.positional.at(0).value() as string;
+    if (eventName !== this.eventName) {
+      this.eventName = eventName;
+      this.shouldUpdate = true;
+    }
+
+    assert(
+      'You must pass a function as the second argument to the `on` modifier',
+      args.positional.at(1) !== undefined && typeof args.positional.at(1).value() === 'function'
+    );
+    let userProvidedCallback = args.positional.at(1).value() as EventListener;
+    if (userProvidedCallback !== this.userProvidedCallback) {
+      this.userProvidedCallback = userProvidedCallback;
+      this.shouldUpdate = true;
+    }
+
+    if (this.shouldUpdate) {
+      let callback = (this.callback = function(this: Element, event) {
```

Comment:
Do you think, it'd be a bit faster to only do this, if `DEBUG && passive || !SUPPORTS_EVENT_OPTIONS && once`?

So kinda like:

```ts
if (DEBUG && passive || !SUPPORTS_EVENT_OPTIONS && once) {
  let callback = (this.callback = function(this: Element, event) {
    if (DEBUG && passive) {
      event.preventDefault = () => {
        assert(
          `You marked this listener as 'passive', meaning that you must not call 'event.preventDefault()': \n\n${userProvidedCallback}`
        );
      };
    }

    if (!SUPPORTS_EVENT_OPTIONS && once) {
      removeEventListener(this, eventName, callback, options);
    }
    return userProvidedCallback.call(null, event);
  });
} else {
  this.callback = userProvidedCallback;
}
```

Or can we neglect this?

Reply:
I was thinking that we didn't care, but honestly its a small change and reduces some overhead so seems good to do. Will update.